mesontest.py

```diff
@@ -284,189 +284,211 @@ def print_stats(self, numlen, tests, name, result, i, logfile, jsonlogfile):
         result_str += "\n\n" + result.get_log()
         if (result.returncode != GNU_SKIP_RETURNCODE) and \
             (result.returncode != 0) != result.should_fail:
-            self.error_count += 1
             if self.options.print_errorlogs:
                 self.collected_logs.append(result_str)
-        logfile.write(result_str)
-        write_json_log(jsonlogfile, name, result)
-
-    def doit(self):
-        if self.is_run:
-            raise RuntimeError('Test harness object can only be used once.')
-        if not os.path.isfile(self.datafile):
-            print('Test data file. Probably this means that you did not run this in the build directory.')
-            return 1
-        self.is_run = True
-        logfilename = self.run_tests(self.options.logbase)
+        if logfile:
+            logfile.write(result_str)
+        if jsonlogfile:
+            write_json_log(jsonlogfile, name, result)
+
+    def print_summary(self, logfile, jsonlogfile):
+        msg = 'Test summary: %d OK, %d FAIL, %d SKIP, %d TIMEOUT' \
+                % (self.success_count, self.fail_count, self.skip_count, self.timeout_count)
```

Comment:
__[flake8]__
* [E126] continuation line over-indented for hanging indent

<!-- Comment by [SideCI](https://sideci.com) -->